src/core/Statistics.cpp

```diff
@@ -548,6 +563,32 @@ Statistics::format_machine_readable(const Config& config,
   return util::join(lines, "");
 }
 
+std::string
+Statistics::format_json(const Config& config,
+                        const util::TimePoint& last_updated) const
+{
+  std::vector<std::string> lines;
+
+  auto add_line = [&](auto id, auto value) {
+    lines.push_back(FMT("\"{}\": {}", id, value));
+  };
+
+  add_line("stats_updated_timestamp", last_updated.sec());
+
+  for (const auto& field : k_statistics_fields) {
+    if (!(field.flags & FLAG_NEVER)) {
+      add_line(field.id, m_counters.get(field.statistic));
+    }
+  }
+
+  add_line("max_cache_size_kibibyte", config.max_size() / 1024);
+  add_line("max_files_in_cache", config.max_files());
+
+  std::sort(lines.begin(), lines.end());
+  std::string result = "{" + util::join(lines, ",") + "}";
```

Comment:
> I'm wondering if the one-line json should end with \n.

Yes, I would prefer ending with a newline. I also wouldn't mind line breaks after statistics lines to make it more human readable.

Reply:
Done! I've also added two spaces of indentation.

Personally I'd actually prefer a one-line "compact" version. Would that justify another `--format` choice, "json-compact" (or "json-pretty" and make "json" compact")? E.g., `jq` has `-c`/`--compact-output` which doesn't pretty-print.

(But I'm also fine with piping through `jq -c` to get my one-liner. :) )

Reply:
Thanks. Let's try to live with one alternative. :slightly_smiling_face: